contracts/access/AccessControlDefaultAdminRules.sol

```diff
@@ -222,19 +282,37 @@ abstract contract AccessControlDefaultAdminRules is IAccessControlDefaultAdminRu
         super._revokeRole(role, account);
     }
 
+    /**
+     * @dev Sets a pending delay into effect if its delay has passed
+     */
+    function _materializeDefaultAdminDelayChange() private {
+        uint48 delaySchedule = _pendingDelaySchedule;
+        if (_isSet(delaySchedule) && _hasPassed(delaySchedule)) _currentDelay = _pendingDelay;
+    }
```

Comment:
I think this function should delete the pending variables as well to leave a consistent state. We can add a boolean return value and we can make both `_materialize` and `_rollback` make sense on their own.

My motivation to propose this is that I was very confused when I saw "rollback delay change" in the body of "begin admin transfer".

```suggestion
    function _materializeDefaultAdminDelayChange() private returns (bool materialized) {
        uint48 delaySchedule = _pendingDelaySchedule;
        if (_isSet(delaySchedule) && _hasPassed(delaySchedule)) {
            _currentDelay = _pendingDelay;
            delete _pendingDelay;
            delete _pendingDelaySchedule;
            return true;
        } else {
            return false;
        }
    }
```

Reply:
I took my time thinking about the best way to split responsibilities and I ended up with a setup that I think comes simple to avoid unnecessary reads in most scenarios, I think you'd like it, but it basically makes this function non-existent.

Can you take a look? I think it's clear enough.

Reply:
We discussed this in a meeting and agreed on the new approach